### DISCUSSION TITLE
Is there a way to export all the data from collection?

### DISCUSSION THREAD

**Discussion by botzill** [directus/directus#6134] [Archive]
Currently, I see that I can export only the current page. Is there a way to export all the data? I have a collection with 10k items and I want to export as CSV.

Thx.

**Comment by benhaynes** [comment]
You can use the export query parameter, and disable the limit by [setting it to `-1`](https://docs.directus.io/reference/api/query/#limit)

https://docs.directus.io/reference/api/query/#export

We are also working on a "backup" endpoint that will export the entire database.

https://github.com/directus/directus/discussions/5107

**Reply by botzill** [reply]
I see.

I think having in the UI possibility to export all the documents would be useful. Same we have an option to apply the filter, a checkbox to export all.

**Reply by botzill** [reply]
Hi @benhaynes 

Btw, seems my custom user token does not work on export endpoint?

I try like this:
```
https://dom.com/items/col?access_token=<custom_token>&export=csv&limit=-1
```
